lib/good_job/systemd_service.rb

```diff
@@ -11,10 +11,10 @@ module GoodJob # :nodoc:
   #
   class SystemdService
     def self.task_observer(_time, _output, thread_error) # :nodoc:
-      return if thread_error.is_a? Concurrent::CancelledOperationError
+      return if thread_error.is_a?(Concurrent::CancelledOperationError) || thread_error.blank?
```

Comment:
Let's put the nil check first:

```suggestion
      return if !thread_error || thread_error.is_a?(Concurrent::CancelledOperationError)
```